doc/adr/0010-data-types-in-columns.md

```diff
@@ -0,0 +1,137 @@
+# 10. Data types in columns
+
+Date: 2023-02-16
+
+## Status
+
+Accepted
+
+## Context
+
+Most Trace Server Protocol (TSP) data structures are currently providing already preformatted strings so that clients can just display the values as. This has the advantage that the clients logic can be limited. However, there are some exceptions. For example, the timestamps are provide as long values and the front-end is using for the time axis. That means timestamps have already special treatment. Because they have dedicated fields in the TSP, the client are aware that they are timestamps and certain actions can be performed. The client code uses it to synchronize the time axis, select time ranges and zoom-in/out operations.
+
+To be able to also, provide additionals actions to select a time range the time range's start and end time need to be provided. For example, some table could provide time ranges in table columns, tool tips for the user to select. However, the client code needs to know how a time range is provided. 
+
+This ADR introduces a `DataType` enum to describe certain data values. The values will be transported a strings and the front-end can use the data type to
+determine the format to display, as well as certain user actions. For type `TIME_RANGE` the client front-end can provide actions to select, navigate and/or zoom to the time range.
+
+### Data structure
+
+The following `DataType` data structure proposed for the TSP is inline with the [DataType enum][data-type] of Trace Compass:
+
+```java
+public enum DataType {
+    /**
+     * Data represent a decimal number
+     */
+    NUMBER,
+    /**
+     * Binary data, where the size orders are powers of 2.
+     */
+    BINARY_NUMBER,
+    /**
+     * Data represent a timestamp in nanoseconds, can be negative
+     */
+    TIMESTAMP,
+    /**
+     * Data represents a duration
+     */
+    DURATION,
+    /**
+     * Data is textual data
+     */
+    STRING,
+    /**
+     * Data representing a time range of string: [start,end],
+     * where `start` and `end` are timestamps in nanoseconds
+     */
+    TIME_RANGE;
+}
+eclipse-cdt-cloud
+```
+
+### Use cases
+
+The first concrete use case for the data types is the `Select min/max time range` in Trace Compass' latency views. New columns can be provided with the min and max time range. The Column descriptor will provide the `data type` and the cell value will transport the range as string `[start,end]`. To select the time range the following user actions can be implemented in the front-end:
+
+- Make the cell entry "clickable", similar to hyperlinks
+  - Advantages:
+    - More discoverable for the user if column is visible on screen and not off screen
+    - Addtional actions can be implemented (CTRL + click or conext-sensitive menu)
+  - Disadvantages:
+    - These columns need to be visible for the feature
```

Comment:
Did you mean to add something after 'for the feature...'?

Reply:
I updated the sentence